_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,596 @@
+Moreover, we have `Oauth`, `Tracing` and also `Basic` client. Each of them are for specific use case. 
+However, we prefer to always use the internal clients because of that platform dependency. 
+Client can be used for instance:
+
+```java
+// 1. Creating the client instance 
+
+BasicExternalKafkaClient basicExternalKafkaClient = new BasicExternalKafkaClient.Builder()
+    .withTopicName("topic-name")
+    .withNamespaceName("namespace-name")
+    .withClusterName("cluster-name")
+    .withKafkaUsername("kafka-username")
+    .withMessageCount(100)
+    .withSecurityProtocol(SecurityProtocol.SSL)
+    .build();
+
+// 2. Sending messages
+
+basicExternalKafkaClient.sendMesssageTls();
+```
+
+## Dependencies
+
+Before we get to writing simple test case in Strimzi system tests, we need to know main dependencies that `system test` module have. 
+I gonna describe the most important one like:
+1. test (kubernetes client)
+2. crd-generator + api (chaining / builder pattern)
+
+### test module
+
+In the test module, we have the Kubernetes client, which we use in the system tests to create, list, delete, update
+Kubernetes resources such as Service, Deployment, Pod, ConfigMap and so on. Basically, we encapsulate the [Kubernetes client](https://github.com/fabric8io/kubernetes-client)
+from fabric8 to use client more handy with not additional and typically redundant method chaining. 
+Moreover, we have our abstraction of Kubernetes cluster and mechanism to detect on which type is currently client. 
+Clusters could be Kubernetes, Minikube and so on.
+
+### crd-generator + api module (not sure about this one...)
+
+In the crd-generator and api modules, we basically use all resources, which we previously described in Resource section. 
+Every time, if the code of Model changes you have to build the whole modules to generate new schema for resources. 
+For instance imagine that you are using `Kafka` resource and you need to create Kafka with the `Cruise Control` 
+support but you have got old schemas, which does not support it.
+You will need to build these modules and generate schema, which will support it and be able to create Kafka with Cruise Control. 
+This modules provides us the fluent way of creating Strimzi custom resources instances such as `Kafka`, `KafkaConnect`, 
+`KafkaConnectS2I`, `KafkaMirrorMaker` and much more.
+
+## How to create a system test
+
+Now, we have all knowledge to create a simple test. There is also some precursors, which i will write it down.
+
+### Precursors
+
+1. Fork the `strimzi-kafka-operator` repository - https://github.com/strimzi/strimzi-kafka-operator
+2. Build the whole project in root directory - `mvn clean install -DskipTests`
+3. Setup your Kubernetes cluster
+4. Login to your Kubernetes cluster
+5. Try to start some of our tests inside `system test` module for instance - `io.strimzi.systemtest.bridge.HttpBridgeST.testSendSimpleMessage()` 
```

Comment:
Shouldn't be pick differet scenario? Like `testSendMessagesTlsAuthenticated`

Reply:
Ok :)